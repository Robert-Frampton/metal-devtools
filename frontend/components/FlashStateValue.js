import Component, {Config} from 'metal-jsx';

import processStateValue from '../lib/processStateValues';
import {debounce, isEqual} from 'lodash';

class FlashStateValue extends Component {
  created() {
    this.removeFlash = debounce(this.removeFlash, 100);

    this._firstRender = true;
  }

  addFlash() {
    this.element.classList.add('flash');

    this.removeFlash();
  }

  removeFlash() {
    setTimeout(() => {
      this.element.classList.remove('flash');
    }, 100);
  }

  syncValue(newVal, oldVal) {
    if (!this._firstRender && !isEqual(newVal, oldVal)) {
      this.addFlash();
    } else {
      this._firstRender = false;
    }
  }

  render() {
    return <span>{processStateValue(this.props.value)}</span>;
  }
}

FlashStateValue.PROPS = {
  value: Config.any()
};

export default FlashStateValue;
