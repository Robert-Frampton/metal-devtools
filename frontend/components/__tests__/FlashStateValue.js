jest.unmock('../FlashStateValue');
jest.unmock('../../lib/processStateValues');

import FlashStateValue from '../FlashStateValue';

describe('FlashStateValue', () => {
	it('should render', () => {
		const component = new FlashStateValue();

		expect(snap(component)).toMatchSnapshot();
	});
});
