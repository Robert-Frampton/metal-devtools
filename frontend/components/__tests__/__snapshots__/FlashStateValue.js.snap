exports[`FlashStateValue should render 1`] = `undefined`;
